### PR TITLE
CIA-270: Refactor Commands

### PR DESCRIPTION
**Pull request by abrown-sg**
This looks like a huge change, but it's mostly moving code around and removing duplication.  I was unable to use Remy's DocOpt pattern mentioned in #76 due to an unresolved DocOpt bug. However this work makes it very easy to switch to his pattern at a later time.

* Refactored commands.py to use a separate class for each command.

* No longer subclassing all commands for static inventory. Removed the redefinition of many methods.

* Cleaned up DocOpt spacing in cli.py to be consistent.

### CODE REVIEW

File: molecule/commands.py [hunk 1]
```diff
@@ -146,8 +144,12 @@ def converge(self, idempotent=False, create_instances=True, create_inventory=Tru
         if self.molecule._state.get('converged'):
             create_inventory = False
 
+        if self.static:
+            create_instances = False
+            create_inventory = False
+
         if create_instances and not idempotent:
-            self.create()
+            Create(self.args, self.molecule).execute()
```
Comment (rgreinho) [inline molecule/commands.py]: Not a blocker, just my 2 cents:  
You used this pattern in a lot of places, but I prefer to see that split in 2 lines, one for creating the object, one for calling the function.

**Comment by rgreinho** [comment]
I like it! Way clearer/simpler! :+1:

**Comment by abrown-sg** [comment]
@rgreinho No problem, I updated to break those all out into multiple lines.

**Comment by retr0h** [comment]
+1 Yeah, deff a good cleanup.

**Pull request merged**